app/controllers/graders_controller.rb

```diff
@@ -177,9 +177,19 @@ def global_actions
       case params[:global_actions]
       when 'assign'
         if grader_ids.blank?
-          render text: I18n.t('assignment.group.select_a_grader'), status: 400
+          render text: I18n.t('assignment.group.select_a_grader'),
+                 status: 400
+        end        
+        if params[:skip_empty_submissions] == 'true'
+          # If the instructor wants to skip empty submissions, remove those groups
+          # from the list of grouping_ids to assign graders to
+          @found_empty_submission = false
```

Comment:
Never use `@` to act as a global variable, this makes code much harder to reason about. In the next line, don't both mutate grouping_ids and assign to it. `find_empty_submissions` is not a very good name for what you're doing, which is filtering out submissions.

Perform two separate operations:
1) Obtain a list of non-empty submission grouping ids in a new variable. Don't mutate grouping_ids. Use `select`.
2) Compare the sizes of the two, and use this to set the value of `found_empty_submission`, which should be a local variable (no @).